internal/core/metadata/rest_device.go

```diff
@@ -81,10 +81,10 @@ func restAddNewDevice(w http.ResponseWriter, r *http.Request) {
 
 	// Service Check
 	// Try by name
-	err = dbClient.GetDeviceServiceByName(&d.Service, d.Service.Service.Name)
+	d.Service, err = dbClient.GetDeviceServiceByName(d.Service.Service.Name)
```

Comment:
`d.Service.Service.Name`

That's lovely. Clearly something needs refactoring here...

Probably out of scope for this PR